Makefile

```diff
@@ -145,21 +165,72 @@ test: build-dirs
 	        ./build/test.sh                                                \
 	    "
 
+.PHONY: build-dirs
 build-dirs:
 	@mkdir -p bin/$(ARCH)
 	@mkdir -p .go/src/$(PKG) .go/pkg .go/bin .go/std/$(ARCH)
 
+.PHONY: clean
 clean: container-clean bin-clean
 
+.PHONY: container-clean
 container-clean:
 	rm -rf .container-* .dockerfile-* .push-*
 
+.PHONY: bin-clean
 bin-clean:
 	rm -rf .go bin
 
+.PHONY: clean-vendor
+## Removes the ./vendor directory.
+clean-vendor:
+	-rm -rf $(VENDOR_DIR)
+
+.PHONY: deps
+## Download build dependencies.
+deps: $(DEP_BIN) $(VENDOR_DIR)
+
+# install dep in a the tmp/bin dir of the repo
+$(DEP_BIN):
+	@echo "Installing 'dep' $(DEP_VERSION) at '$(DEP_BIN_DIR)'..."
+	mkdir -p $(DEP_BIN_DIR)
+ifeq ($(UNAME_S),Darwin)
+	@curl -L -s https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-darwin-amd64 -o $(DEP_BIN)
+	@cd $(DEP_BIN_DIR) && \
+	echo "1544afdd4d543574ef8eabed343d683f7211202a65380f8b32035d07ce0c45ef  dep" > dep-darwin-amd64.sha256 && \
+	shasum -a 256 --check dep-darwin-amd64.sha256
+else
+	@curl -L -s https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-linux-amd64 -o $(DEP_BIN)
+	@cd $(DEP_BIN_DIR) && \
+	echo "31144e465e52ffbc0035248a10ddea61a09bf28b00784fd3fdd9882c8cbb2315  dep" > dep-linux-amd64.sha256 && \
+	sha256sum -c dep-linux-amd64.sha256
+endif
+	@chmod +x $(DEP_BIN)
+
+$(VENDOR_DIR): Gopkg.toml Gopkg.lock
+	@echo "checking dependencies..."
+	@$(DEP_BIN) ensure -v
+
 GOFORMAT_FILES := $(shell find  . -name '*.go')
 
-.PHONY: format ## Formats any go file that differs from gofmt's style and removes uneeded imports
+.PHONY: format ## Formats any go file that differs from gofmt's style and removes unused imports
 format: 
 	@gofmt -s -l -w ${GOFORMAT_FILES}
-	@goimports -l -w ${GOFORMAT_FILES}
\ No newline at end of file
+	@goimports -l -w ${GOFORMAT_FILES}
+
+.PHONY: tools
+tools: ## Installs required go tools
+	@go get -u github.com/alecthomas/gometalinter && gometalinter --install
+	@go get -u golang.org/x/tools/cmd/goimports
+	
+.PHONY: install
+install: ## Fetches all dependencies using dep
+	dep ensure -v
```

Comment:
As `dep` is not in environment PATH, I think `./tmp/bin/dep ensure -v` will work.